pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -295,7 +311,17 @@ type UpdatePlainStrategy struct {
 func (c UpdatePlainStrategy) Op() ClusterChangeApplyStrategyOp { return updateStrategyPlainAnnValue }
 
 func (c UpdatePlainStrategy) Apply() error {
-	updatedRes, err := c.aou.identifiedResources.Update(c.newRes)
+	var updatedRes ctlres.Resource
+	var err error
+
+	if c.aou.opts.ServerSideApply {
+		updatedRes, err = ctlres.WithIdentityAnnotation(c.newRes, func(r ctlres.Resource) (ctlres.Resource, error) {
+			resBytes, _ := r.AsYAMLBytes()
+			return c.aou.identifiedResources.Patch(r, types.ApplyPatchType, resBytes, ctlres.PatchOpts{DryRun: false})
+		})
+	} else {
+		updatedRes, err = c.aou.identifiedResources.Update(c.newRes)
```

Comment:
You are right. `tryToResolveUpdateConflict` recalculates Change from a new existing resource. In case of SSA only conflict we can have is when we are conflicting with another fieldManager, no amount of new diffing going to help here.